### DISCUSSION TITLE
Chrome and Edge works fine but Firefox gives "codecs not supported by client"

### DISCUSSION THREAD

**Discussion by joshting** [bluenviron/mediamtx#3710] [Questions] [answered]
### Question

Please help. The media router works in routing video streams via WebRTC to Chrome and Edge browsers but when I use Firefox (latest version 126), I get "codecs not supported by client" error.

**Comment by asturl** [comment]
What IS the codec you're streaming? 
If it's H264 (which is likely, given how narrow the codec support for WebRTC is) then it should only be an issue on **Android** Firefox, but more details would help.

You can type "about:support" into Firefox's address bar and scroll down to the Media table to see which codecs your Firefox supports, and maybe we can find some problem there

**Reply by joshting** [reply]
This is what I got, I can't see there is any problem here if H264 is what WebRTC requires.  BTW, I use the browser client code https://github.com/bluenviron/mediamtx/blob/main/internal/servers/webrtc/read_index.html as reference in my Angular code.  Works perfectly fine in Chrome.

| Codec Name	|Software Decoding | Hardware Decoding |
| --- | --- | --- |
| H264 | Supported | Unsupported |
| VP9 | Supported | Unsupported |
| VP8 | Supported | Unsupported |
| AV1 | Supported | Unsupported |
| HEVC | Unsupported | Unsupported |
| Theora | Supported | Unsupported |
| AAC | Supported | Unsupported |
| MP3 | Supported | Unsupported |
| Opus | Supported | Unsupported |
| Vorbis | Supported | Unsupported |
| FLAC | Supported | Unsupported |
| Wave | Supported | Unsupported |

**Comment by Falkitop** [comment, marked as answer]
I fixed it by setting `media.webrt.hw.h264.enabled` in either about:config or prefs.js via ssh

**Comment by Falkitop** [comment]
Another Tip is to not forget to install / include ffmpeg. Otherwise the screen wont display anything. Firefox will automatically find ffmpeg if its installed correctly. the OpenWebRTC Plugin from Cisco is not needed I assume
